### ISSUE TITLE
After upgrading to v7.0.1 from 6.2.2 getPurchaseHistory does not return response

### ISSUE THREAD

**Issue by SergiOnGit** [dooboolab/react-native-iap#1409]
> Please use [Discussion board](https://github.com/dooboolab/react-native-iap/discussions) if you want get some help out of it. Please report issue if there is a bug.

### Version of react-native-iap
7.0.1

### Version of react-native
0.64.2

### Platforms you faced the error (IOS or Android or both?)
Android. Not checked on IOS

### Expected behavior
It should return purchase history array

### Actual behavior
It doesn't return anything

### Tested environment (Emulator? Real Device?)
Both

### Steps to reproduce the behavior
Just upgrade to this version and call getPurchaseHistory()

What am i missing? It is working fine in 6.2.2.

**Comment by Montezi** [comment]
Hi! I have the same issue  with almost same enviroment , with react native 0.63.4.

**Comment by nikmedoed** [comment]
The same problem. `getAvailablePurchases()` also doesn't work.
`RN 0.59.8`, android

Workin on 6.3.0

Rolling back again

**Comment by marcpechaitis** [comment]
Same here (on android device, works fine on iOS):
```
    const purchaseHistory = await RNIap.getPurchaseHistory();
    console.log('purchaseHistory', purchaseHistory);
    purchaseHistory.forEach((purchase) => {
      console.log('🔷🔷🔷🔷🔷 history', purchase.productId);
    });
 ```

`await RNIap.getPurchaseHistory();` never resolves/rejects, and the `console.log('purchaseHistory', purchaseHistory);` does not get called in 7.0.1

It works great in 6.3.0:
```
purchaseHistory [{"dataAndroid": <redacted>}]
[Fri Jul 09 2021 08:34:17.580]  LOG      🔷🔷🔷🔷🔷 history bingo.consumable.one
[Fri Jul 09 2021 08:34:17.670]  LOG      🔷🔷🔷🔷🔷 history bingo.consumable.two
[Fri Jul 09 2021 08:34:17.700]  LOG      🔷🔷🔷🔷🔷 history bingo.consumable.three
[Fri Jul 09 2021 08:34:17.700]  LOG      🔷🔷🔷🔷🔷 history bingo.nonconsumable.four
```

**Comment by andresesfm** [comment]
I'm assuming you are all calling `initConnection()` beforehand, correct?. 
Also can somebody please post the adb logcat? I'm happy to take a look

**Comment by marcpechaitis** [comment]
@andresesfm -- `initConnection()` is invoked automatically by using the `useIap()` hook, right?

In any event, I made no changes to my code... i simply upgraded react-native-iap version from 6.3.0 (where it works) to 7.0.1 (where it does not)

**Comment by hyochan** [comment]
> @andresesfm -- `initConnection()` is invoked automatically by using the `useIap()` hook, right?
> 
> In any event, I made no changes to my code... i simply upgraded react-native-iap version from 6.3.0 (where it works) to 7.0.1 (where it does not)

@marcpechaitis You are right. `initConnection` invoked internally in `useIap` hook. However, I don't see any breaking changes on hooks between two versions.

Could you kindly give additional information that we can have to help you out. As @andresesfm pointed out, any logcat or logs will be helpful.

**Comment by SergiOnGit** [comment]
@hyochan getPurchaseHistory() is fixed, but now I noticed that the same happens with getAvailablePurchases(). Please fix it too.

**Comment by hyochan** [comment]
> @hyochan getPurchaseHistory() is fixed, but now I noticed that the same happens with getAvailablePurchases(). Please fix it too.

Thanks for the report! I've made the fixes on `7.0.3`.

**Comment by SergiOnGit** [comment]
@hyochan It is fixed. Thank you!
